File: charcorba-c++/Client/src/standard_impl.cpp
#include <standard_impl.h>
#include <string>
#include <map>
#include <iostream>
#include <chatroom.h>
#include <client.h>
#include <tableau_blanc.h>

using namespace std ;
// Implementation for interface Standard
Standard_impl * Standard_impl::m_static_standard = 0 ;

Standard_impl::Standard_impl()
{
	Standard_impl::m_static_standard = this ;
}

CORBA::Boolean
Standard_impl::afficher_message(const char* pseudo,const char* message)
  throw(
    ::CORBA::SystemException)
{
	cout << "[DEBUG - Standard]\tAffichage du message '" << message << "' venant de '" << pseudo << "'" << endl ;
	CORBA::Boolean retval ;
	retval = true;
    Client::get_instance()->message_recu(QString(pseudo),QString(message));
	return retval; 
}

CORBA::Boolean
Standard_impl::inviter_client_chatroom ( const char* chatroom )
  throw(
    ::CORBA::SystemException)

{
	cout << "[DEBUG - Standard]\tRéception d'une invitation à participer au chat '" << chatroom << "'" << endl ;
	CORBA::Boolean retval = true ;
	
	CORBA::Object_var service = Client::get_instance()->m_MICO_ORB->connecter_service(chatroom);
	
	Chatroom_var service_chatroom ;
	service_chatroom = Chatroom::_narrow(service.in()) ;

	if (CORBA::is_nil(service_chatroom))
	{
		cerr << "[DEBUG]\tL'IOR n'est pas une référence sur un service." << endl;
	}
	
	Client::get_instance()->m_liste_chatrooms_distantes.insert( pair<string,Chatroom_var>(string(chatroom),service_chatroom));
	
	QString tmp = QString::fromStdString(chatroom);
	Client::get_instance()->signal_invitation_chatroom(tmp);
	return retval;
}

CORBA::Boolean
Standard_impl::inviter_client_tableaublanc ( const char* tableau )
  throw(
    ::CORBA::SystemException)
{
	cout << "[DEBUG - Standard]\tRéception d'une invitation à participer au tableau blanc '" << tableau << "'" << endl ;
	
	CORBA::Boolean retval = true ;
	
	CORBA::Object_var service = Client::get_instance()->m_MICO_ORB->connecter_service(tableau);
	
	TableauBlanc_var service_tableaublanc ;
	service_tableaublanc = TableauBlanc::_narrow(service.in()) ;

	if (CORBA::is_nil(service_tableaublanc))
	{
		cerr << "[DEBUG]\tL'IOR n'est pas une référence sur un service." << endl;
	}
	
	Client::get_instance()->m_liste_tableauxblancs_distants.insert( pair<string,TableauBlanc_var>(string(tableau),service_tableaublanc));
	
	QString tmp = QString::fromStdString(tableau);
	Client::get_instance()->participer_tableau_blanc(tmp);
	
	return retval;
}

CORBA::Boolean
Standard_impl::signal_chatroom( const char* chatroom, CORBA::ULong idpixel )
  throw(
    ::CORBA::SystemException)
{
	CORBA::Boolean retval;
	Client::get_instance()->signal_chatroom(chatroom);
	return retval;
}

CORBA::Boolean
Standard_impl::signal_tableaublanc( const char* tableau, CORBA::ULong idpixel )
  throw(
    ::CORBA::SystemException)

{
  CORBA::Boolean retval;
  Client::get_instance()->sync_tableau_blanc (QString (tableau),idpixel);
  return retval;
}


CORBA::Boolean
Standard_impl::ping()
  throw(
    ::CORBA::SystemException)

{
	CORBA::Boolean retval = true ;
	return retval;
}

File: charcorba-java/Client/Standard_impl.java
package Client;

import org.omg.CosNaming.NamingContextPackage.CannotProceed;
import org.omg.CosNaming.NamingContextPackage.InvalidName;
import org.omg.CosNaming.NamingContextPackage.NotFound;

import chatroom.Chatroom;
import chatroom.ChatroomHelper;
import orb_pkge.COrb;
import tableau_blanc.TableauBlanc;
import tableau_blanc.TableauBlancHelper;

public class Standard_impl extends StandardPOA
{
	public static Standard_impl static_standard;
	
	public Standard_impl()
	{
		static_standard = this;
	}
	
	public boolean afficher_message(String pseudo_client, String message) 
	{
		boolean retval = true;
	    Client.singleton_client.message_recu(pseudo_client,message);
		return retval; 
	}
	
	public boolean inviter_client(String chatroom) 
	{
		boolean retval = true ;
		
		org.omg.CORBA.Object service = null;
		try {
			service = COrb.static_orb.connecter_service(chatroom);
		} catch (NotFound e) {
			// TODO Bloc catch auto-généré
			e.printStackTrace();
		} catch (CannotProceed e) {
			// TODO Bloc catch auto-généré
			e.printStackTrace();
		} catch (InvalidName e) {
			// TODO Bloc catch auto-généré
			e.printStackTrace();
		}
		
		Chatroom service_chatroom = ChatroomHelper.narrow(service) ;

		Client.singleton_client.m_liste_chatrooms_distantes.put(chatroom,service_chatroom);
		
		Client.singleton_client.signal_invitation_chatroom(chatroom);
		return retval;
	}

	public boolean signal_chatroom(String chatroom) 
	{
		boolean retval = true;
		Client.singleton_client.signal_chatroom(chatroom, 0);
		return retval;
	}
	
	public boolean ping() 
	{
		boolean retval = true;
		return retval;
	}

	public boolean signal_chatroom(String chatroom, int idpixel) 
	{
		boolean retval = true;
		
		Client.singleton_client.signal_chatroom(chatroom, idpixel);
		return retval;
	}

	public boolean signal_tableaublanc(String tableau, int idpixel) 
	{
		  boolean retval = true;
		  Client.singleton_client.sync_tableau_blanc (tableau,idpixel);
		  return retval;
	}
	
	public boolean inviter_client_tableaublanc ( String tableau )
	{	
		boolean retval = true ;
		org.omg.CORBA.Object service = null;
		try {
			service = COrb.static_orb.connecter_service(tableau);
		} catch (NotFound e) {
			// TODO Bloc catch auto-généré
			e.printStackTrace();
		} catch (CannotProceed e) {
			// TODO Bloc catch auto-généré
			e.printStackTrace();
		} catch (InvalidName e) {
			// TODO Bloc catch auto-généré
			e.printStackTrace();
		}
		
		TableauBlanc service_tableaublanc ;
		service_tableaublanc = TableauBlancHelper.narrow(service) ;

		Client.singleton_client.m_liste_tableauxblancs_distants.put(tableau,service_tableaublanc);
		Client.singleton_client.participer_tableau_blanc(tableau);
		return retval;
	}

	public boolean inviter_client_chatroom(String chatroom) 
	{
		boolean retval = true ;
		
		org.omg.CORBA.Object service = null;
		try {
			service = COrb.static_orb.connecter_service(chatroom);
		} catch (NotFound e) {
			// TODO Bloc catch auto-généré
			e.printStackTrace();
		} catch (CannotProceed e) {
			// TODO Bloc catch auto-généré
			e.printStackTrace();
		} catch (InvalidName e) {
			// TODO Bloc catch auto-généré
			e.printStackTrace();
		}
		
		Chatroom service_chatroom = ChatroomHelper.narrow(service) ;

		Client.singleton_client.m_liste_chatrooms_distantes.put(chatroom,service_chatroom);
		
		Client.singleton_client.signal_invitation_chatroom(chatroom);
		return retval;
	}

}
